### PR TITLE
Improve tests

### PR DESCRIPTION
**Pull request by mconcas**

### CODE REVIEW

**Comment by mconcas** [comment]
Ok, here there is a pedantic and very granular st of tests. This is a purposal, if you all agree I'll make the 12 Test package `required`.

To run them locally:
```tests/run_tests.sh```
To run few of them:
```tests/run_tests.sh [test1 [test2 [test3]]]```
Rigidity and parameters of the test are up to you. Let me know what you think
@fgrosa  @fcatalan92 @fmazzasc @pfecchio @mpuccio @cterrevo

File: .pylintrc [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+indent-string='    '
+max-line-length=100
+
+[BASIC]
+function-naming-style=camelCase
+argument-naming-style=camelCase
+variable-naming-style=camelCase
+method-naming-style=camelCase
+attr-naming-style=camelCase
+
```
Comment (fcatalan92) [inline .pylintrc]: Maybe we could relax the design requirements, e.g. fixing a larger maximum number of function arguments or local variables and so on
```
[DESIGN]
max-args=10
max-locals=40
....
```
In my experience methods that exceeds the default configuration could be fine/necessary. @hipe4ml/developers What do you think?

**Comment by fcatalan92** [comment]
@mconcas To me the very granular set of tests is fine. However, I do not get the difference between pylint and flake8 and if both tests are steered by `.pylintrc`

File: .pylintrc (hunk 1 above)
Comment (mconcas) [reply .pylintrc]: > @mconcas To me the very granular set of tests is fine. However, I do not get the difference between pylint and flake8 and if both tests are steered by `.pylintrc`

They have a great region of overlap, with the first run they both raised 5 errors each different each other. Eventually we could also consider to remove one of them, for the moment I would try to give them a chance.
AFAIK only pylint relies on `.pylintrc`

**Comment by mconcas** [comment]
> @mconcas To me the very granular set of tests is fine. However, I do not get the difference between pylint and flake8 and if both tests are steered by `.pylintrc`

They have a great region of overlap, with the first run they both raised 5 errors each different each other. Eventually we could also consider to remove one of them, for the moment I would try to give them a chance.
AFAIK only pylint relies on `.pylintrc`

File: .pylintrc [hunk 2]
```diff
@@ -0,0 +1,13 @@
+[FORMAT]
+indent-string='    '
+max-line-length=100
```
[the remark this answers is not in the document]
Comment (mconcas) [reply .pylintrc]: Fine to me

File: .pylintrc [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+[BASIC]
+function-naming-style=camelCase
+argument-naming-style=camelCase
+variable-naming-style=camelCase
+method-naming-style=camelCase
+attr-naming-style=camelCase
+
+[MESSAGES CONTROL]
+disable=missing-docstring,useless-object-inheritance
```
[the remark this answers is not in the document]
Comment (mconcas) [reply .pylintrc]: True, although this can slow the work very much.

File: .pylintrc (hunk 1 above)
Comment (mconcas) [reply .pylintrc]: Might definitely be the case. Not a strong opinion.

**Pull request merged**